jerry-core/parser/js/js-parser.c

```diff
@@ -2236,19 +2236,62 @@ parser_parse_script (const uint8_t *source_p, /**< source code */
                      bool is_strict, /**< strict mode */
                      ecma_compiled_code_t **bytecode_data_p) /**< [out] JS bytecode */
 {
-  parser_error_location parse_error;
-  *bytecode_data_p = parser_parse_source (source_p, size, is_strict, &parse_error);
+  parser_error_location_t parser_error;
+  *bytecode_data_p = parser_parse_source (source_p, size, is_strict, &parser_error);
 
   if (!*bytecode_data_p)
   {
-    if (parse_error.error == PARSER_ERR_OUT_OF_MEMORY)
+    if (parser_error.error == PARSER_ERR_OUT_OF_MEMORY)
     {
       /* It is unlikely that memory can be allocated in an out-of-memory
        * situation. However, a simple value can still be thrown. */
       return ecma_make_error_value (ecma_make_simple_value (ECMA_SIMPLE_VALUE_NULL));
     }
+#if JERRY_ENABLE_ERROR_MESSAGES
+    const char *err_str_p = parser_error_to_string (parser_error.error);
+    uint32_t err_str_size = lit_zt_utf8_string_size ((const lit_utf8_byte_t *) err_str_p);
 
-    return ecma_raise_syntax_error (parser_error_to_string (parse_error.error));
+    char line_str_p[ECMA_MAX_CHARS_IN_STRINGIFIED_UINT32];
+    uint32_t line_len = ecma_uint32_to_utf8_string (parser_error.line,
+                                                    (lit_utf8_byte_t *) line_str_p,
+                                                    ECMA_MAX_CHARS_IN_STRINGIFIED_UINT32);
+
+    char col_str_p[ECMA_MAX_CHARS_IN_STRINGIFIED_UINT32];
+    uint32_t col_len = ecma_uint32_to_utf8_string (parser_error.column,
+                                                   (lit_utf8_byte_t *) col_str_p,
+                                                   ECMA_MAX_CHARS_IN_STRINGIFIED_UINT32);
+
+    uint32_t msg_size = err_str_size + line_len + col_len + (uint32_t) sizeof (" [line: , column: ]");
+
+    ecma_value_t error_value = ecma_make_simple_value (ECMA_SIMPLE_VALUE_EMPTY);
+
+    JMEM_DEFINE_LOCAL_ARRAY (error_msg_p, msg_size, char);
+    char *err_msg_pos_p = error_msg_p;
+
+    strncpy (err_msg_pos_p, err_str_p, err_str_size);
+    err_msg_pos_p += err_str_size;
+
+    strncpy (err_msg_pos_p, " [line: ", 8);
+    err_msg_pos_p += 8;
```

Comment:
There are a bit too much hard-coded string and length constants around here that have to be kept in sync manually and easy to get wrong. (IMHO) Something like:
```c
#define POS_START  " [line: "
#define POS_MIDDLE ", column: "
#define POS_END    "]"
uint32_t msg_size = err_str_size + line_len + col_len + (uint32_t) sizeof (POS_START) + (uint32_t) sizeof (POS_MIDDLE) + (uint32_t) sizeof (POS_END);
strncpy (err_msg_pos_p, POS_START, sizeof (POS_START));
err_msg_pos_p += sizeof (POS_START);
// etc...
```
might be easier to maintain.